### PR TITLE
[Feature] Agency ArangoSync State check

### PR DESCRIPTION
**Pull request by ajanikow**

### CODE REVIEW

**Review by nikita-vanyasin — APPROVED** [review]

File: pkg/deployment/agency/arangosync_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	require.True(t, s.Agency.ArangoDB.ArangoSync.ArangoSync.State.Outgoing.Targets.Exists())
+	require.True(t, s.Agency.ArangoDB.ArangoSync.IsSyncInProgress())
+}
+
+func Test_Sync_Source(t *testing.T) {
+	var s DumpState
+	require.NoError(t, json.Unmarshal(syncTarget, &s))
+
+	require.NotNil(t, s.Agency.ArangoDB.ArangoSync.ArangoSync)
+	require.NotNil(t, s.Agency.ArangoDB.ArangoSync.ArangoSync.State.Incoming.State)
```
Comment (informalict) [inline pkg/deployment/agency/arangosync_test.go:44]: why do we have here ArangoSync twice?

File: pkg/deployment/agency/state.go [hunk 2]
```diff
@@ -87,13 +87,18 @@ func (c *cache) loadState(ctx context.Context, client agency.Agency) (State, err
...
+	Arango   State   `json:"arango"`
+	ArangoDB StateDB `json:"arangodb,omitempty"`
 }
 
 type DumpState struct {
 	Agency StateRoot `json:"agency"`
 }
 
+type StateDB struct {
+	ArangoSync ArangoSyncLazy `json:"arangoSync,omitempty"`
```
Comment (informalict) [inline pkg/deployment/agency/state.go:99]:
```suggestion
	ArangoSync ArangoSyncLazy `json:"arangosync,omitempty"`
```

**Review by informalict — COMMENTED** [review]
comments

File: pkg/deployment/agency/arangosync_test.go (hunk 1 above)
Comment (ajanikow) [reply pkg/deployment/agency/arangosync_test.go:44]: It is due to Lazy loader - in case of failure I do not want to break Operator Deployment loop

**Review by informalict — APPROVED** [review]
LGTM

**Pull request merged**